framework/src/main/java/org/checkerframework/framework/util/dependenttypes/DependentTypesError.java

```diff
@@ -27,7 +27,7 @@ public class DependentTypesError {
 
     /** Regular expression for unparsing string representations of this class (gross). */
     private static final Pattern ERROR_PATTERN =
-            Pattern.compile("\\[error for expression: (.*); error: (.*)\\]");
+            Pattern.compile("(?s)\\[error for expression: (.*); error: (.*)\\]");
```

Comment:
See earlier comment about using [DOTALL](https://docs.oracle.com/javase/8/docs/api/java/util/regex/Pattern.html#DOTALL) to make the code easier to read.

Reply:
And think about whether we need to use DOTALL at all in these. Maybe in some we do.

Reply:
Can I ask why would you suggest some dotall is unnecessary?

Reply:
Does DOTALL make a difference only for Windows? If not, if DOTALL wasn't required for Unix, why is it required for Windows? These patterns generally seem to be on single lines, so it is not obvious why DOTALL is required now.

Reply:
I've replaced the Dotall with withspace check, which checks for only the places that are necessary. How do you like it?